crates/router/src/connector/billwerk/transformers.rs

```diff
@@ -30,97 +31,216 @@ impl<T>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
-        match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
+    fn try_from(
+        item: types::ResponseRouterData<
+            api::PaymentMethodToken,
+            BillwerkTokenResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: Ok(types::PaymentsResponseData::TokenizationResponse {
+                token: item.response.id.expose(),
             }),
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
-        }
+            ..item.data
+        })
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
+
+#[derive(Debug, Serialize)]
+pub struct BillwerkCustomerObject {
+    handle: Option<String>,
+    email: Option<Email>,
+    address: Option<Secret<String>>,
+    address2: Option<Secret<String>>,
+    city: Option<String>,
+    country: Option<common_enums::CountryAlpha2>,
+    first_name: Option<Secret<String>>,
+    last_name: Option<Secret<String>>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct BillwerkPaymentsRequest {
+    handle: String,
+    amount: i64,
+    source: Secret<String>,
+    currency: common_enums::Currency,
+    customer: BillwerkCustomerObject,
+    metadata: Option<SecretSerdeValue>,
+    settle: bool,
+}
+
+impl TryFrom<&BillwerkRouterData<&types::PaymentsAuthorizeRouterData>> for BillwerkPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &BillwerkRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let source = match item.router_data.get_payment_method_token()? {
+            types::PaymentMethodToken::Token(pm_token) => Ok(Secret::new(pm_token)),
+            _ => Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "payment_method_token",
+            }),
+        }?;
+        Ok(Self {
+            handle: item.router_data.connector_request_reference_id.clone(),
+            amount: item.amount,
+            source,
+            currency: item.router_data.request.currency,
+            customer: BillwerkCustomerObject {
+                handle: item.router_data.customer_id.clone(),
+                email: item.router_data.request.email.clone(),
+                address: item.router_data.get_optional_billing_line1(),
+                address2: item.router_data.get_optional_billing_line2(),
+                city: item.router_data.get_optional_billing_city(),
+                country: item.router_data.get_optional_billing_country(),
+                first_name: item.router_data.get_optional_billing_first_name(),
+                last_name: item.router_data.get_optional_billing_last_name(),
```

Comment:
if billing address and email are passed as None, will the payment instrument tokenized against customer_id alone? 

And will PG allow tokenization of same payment instrument against a customer_id multiple times?

Reply:
Pg will create a customer itself and create a payment token. and same customer_id can be used multiple times as it is short-lived token